pyproject.toml

```diff
@@ -133,18 +193,37 @@ use_parentheses = true
 ensure_newline_before_comments = true
 line_length = 88
 
-[tool.black]
-# Use extend-exclude so that the default excludes (including those from .gitignore)
-# are not overwritten.
-extend-exclude = "docs/sphinx-starter-pack"
-
 [tool.codespell]
+ignore-words-list = "buildd,crate,keyserver,comandos,ro,dedent,dedented"
+skip = ".tox,.git,build,.*_cache,__pycache__,*.tar,*.snap,*.png,./node_modules,./docs/_build,.direnv,.venv,venv,.vscode"
 quiet-level = 3
-skip = "./docs/_build,.direnv,.git,.mypy_cache,.pytest_cache,.venv,__pycache__,venv"
-ignore-words-list = "warmup,buildd,astroid"
+check-filenames = true
+
+[tool.pytest.ini_options]
+minversion = "7.0"
+testpaths = "tests"
+xfail_strict = true
+markers = ["slow: slow tests"]
+
+[tool.coverage.run]
+branch = true
+omit = ["tests/**"]
+
+[tool.coverage.report]
+skip_empty = true
+exclude_also = [
+    "if (typing\\.)?TYPE_CHECKING:",
+]
 
 [tool.pyright]
-ignore = [
+strict = ["starcraft"]
```

Comment:
```suggestion
strict = ["rockcraft"]
```

Reply:
thanks for this! Turns out we had a bunch of pyright issues to address